templates/_worker.js/index.ts

```diff
@@ -91,11 +94,11 @@ export default {
 		for (const route of routes) {
 			if (
 				'middlewarePath' in route &&
-				route.middlewarePath in __BUILD_OUTPUT__
+				(route.middlewarePath as string) in __BUILD_OUTPUT__
 			) {
-				const item = __BUILD_OUTPUT__[route.middlewarePath];
+				const item = __BUILD_OUTPUT__[route.middlewarePath as string];
```

Comment:
again, I would avoid unnecessary/confusing casting and would go with something like this instead:
![Screenshot 2023-04-18 at 10 05 42](https://user-images.githubusercontent.com/61631103/232729407-6bcb046e-8eab-4f61-a611-08492e6c35a8.png)